packages/medusa/src/services/order-edit.ts

```diff
@@ -467,4 +404,36 @@ export default class OrderEditService extends TransactionBaseService {
       return orderEdit
     })
   }
+
+  protected async deleteClonedItems(orderEditId: string): Promise<void> {
+    const manager = this.transactionManager_ ?? this.manager_
+    const lineItemServiceTx = this.lineItemService_.withTransaction(manager)
+    const lineItemAdjustmentServiceTx =
+      this.lineItemAdjustmentService_.withTransaction(manager)
+    const taxProviderServiceTs =
+      this.taxProviderService_.withTransaction(manager)
+
+    const clonedLineItems = await lineItemServiceTx.list(
+      {
+        order_edit_id: orderEditId,
+      },
+      {
+        select: ["id", "tax_lines", "adjustments"],
+        relations: ["tax_lines", "adjustments"],
+      }
+    )
+    const clonedItemIds = clonedLineItems.map((item) => item.id)
+
+    await Promise.all([
+      clonedItemIds.map((id) => {
+        return lineItemServiceTx.delete(id)
+      }),
+      taxProviderServiceTs.clearLineItemsTaxLines(clonedItemIds),
+      clonedItemIds.map((id) => {
+        return lineItemAdjustmentServiceTx.delete({
+          item_id: id,
+        })
+      }),
+    ])
```

Comment:
thought: since there are foreign keys from LineItem to Adjustments and TaxLines I think we need to ensure that Adjustments and TaxLines are deleted first. Othwerwise, we risk getting a foreign key constraint error